src/inventory_ui.cpp

```diff
@@ -86,9 +86,12 @@ item_name_t &get_cached_name( item const *it )
 {
     auto iter = item_name_cache.find( it );
     if( iter == item_name_cache.end() ) {
+        const std::string name = remove_color_tags( it->tname( 1, false, 0, true, false ) );
+        const std::string name_full = string_format( "%s%d%d",
+                                      remove_color_tags( it->tname( 1, true, 0, true, false ) ),
+                                      it->max_link_length(), it->link_length() );
```

Comment:
Isn't this redundant? The items that have a link up already include link length/max length in their name and the other ones get stacked so sorting never comes into play.

Reply:
The length/max_length aren't actually a part of tname, they're an inventory-only addition to the name like ammo/max_ammo, so I need to add them to the internal name so the inventory only consolidates cables with the same values. Maybe I should add a comment explaining that?

Reply:
Ok, I thought the cable length bit was in `tname()` instead of `display_name()`. But still:

You're handling cable lengths for stacking [here](https://github.com/CleverRaven/Cataclysm-DDA/blob/823fa0be255841a2d24dd89ca42f83276c40ce0f/src/inventory_ui.cpp#L1280-L1281) and for collation [here](https://github.com/CleverRaven/Cataclysm-DDA/blob/823fa0be255841a2d24dd89ca42f83276c40ce0f/src/inventory_ui.cpp#L1398-L1399C58).

This cached name is only used as a sort key and has no impact on the above two. If you want to sort by remaining length, then this is the wrong place for it because it's ASCII sorting so it looks like this
![Screenshot from 2023-07-21 20-25-03](https://github.com/CleverRaven/Cataclysm-DDA/assets/68240139/a3a876b6-efb1-44a9-827e-926ad09e304c)
It can be done properly in the preset's `sort_compare()`.